controllers/jenkins/pipelinerun/pipelinerun_controller.go

```diff
@@ -39,12 +49,175 @@ type Reconciler struct {
 // Reconcile is part of the main kubernetes reconciliation loop which aims to
 // move the current state of the cluster closer to the desired state.
 func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-	_ = context.Background()
-	r.Log.WithValues("pipelinerun", req.NamespacedName).Info("Hello PipelineRun")
+	ctx := context.Background()
+	log := r.Log.WithValues("pipelinerun", req.NamespacedName)
+
+	// get pipeline run
+	var pr devopsv1alpha4.PipelineRun
+	if err := r.Client.Get(ctx, req.NamespacedName, &pr); err != nil {
+		log.Error(err, "unable to fetch PipelineRun")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// get pipeline
+	pipelineRef := v1.GetControllerOf(&pr)
+
+	if pipelineRef == nil {
+		log.Error(nil, "skipped to reconcile this pipeline run due to not found pipeline reference in pipelinerun owner references.")
```

Comment:
I prefer to change `pipeline run` to `PipelineRun`. We should keep using a consistent expression.

`pipelinerun` to `PipelineRun`

Reply:
It's weird.

First, you think this is an error, but you didn't return the error.
Second, why do you give it a `nil`? `log.Error(nil`. Please reconsider if it's an error. And try to find a better statement.